addons/binding/org.openhab.binding.connectsdk/src/main/java/org/openhab/binding/connectsdk/handler/AbstractChannelHandler.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.connectsdk.handler;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.connectsdk.device.ConnectableDevice;
+import com.connectsdk.service.capability.listeners.ResponseListener;
+import com.connectsdk.service.command.ServiceCommandError;
+import com.connectsdk.service.command.ServiceSubscription;
+
+/**
+ * An abstract implementation of ChannelHander which serves as a base class for all concrete instances.
+ *
+ * @author Sebastian Prehn
+ * @since 1.8.0
+ */
+public abstract class AbstractChannelHandler<T> implements ChannelHandler {
```

Comment:
Are you sure this is an abstract class? I don't see abstract methods, but do see default empty implementations. It suggests a base class instead.

Reply:
It does not make sense to instantiate the class, thus I declared it abstract. Your observation is correct. It is also a base class. I would like to keep the abstract key word in front of it. Are you suggesting to change the name of this class to "BaseChannelHandler"?

Reply:
That's probably a better name (although I don't know what preferred naming convention is). There are other ways to make a class less accessible. Make de class package private by removing the public keyword on the class. That way only classes inside the package can use the class. Also you can add a protected constructor. That way only super classes can instantie the class.

Reply:
Will make it package private and rename to BaseChannelHandler. It still needs to be abstract as not all Interface methods are implemented. Those must be implemented by the actual handlers.